### PR TITLE
[CPM-79] student route api tests

### PR DESCRIPTION
**Pull request by wryao64**
wrote tests for student-route; changed send() to json() for all endpoints for consistency

### CODE REVIEW

**Review by HongShi10 — APPROVED** [review]
LGTM maybe have more smaller commits next time

File: api/src/routes/course-route.js [hunk 1]
```diff
@@ -196,7 +197,7 @@ router.delete("/:courseCode", async (req, res) => {
         } else {
             if (course === null) {
                 LOGGER.error("Course does not exist");
-                res.status(204).send();
+                res.status(204).json({ msg: "Requested object not found" });
```
Comment (HongShi10) [inline api/src/routes/course-route.js:200]: change this to 404

**Review by HongShi10 — CHANGES_REQUESTED** [review]

**Review by daltonlim — APPROVED** [review]
LGTM

**Review by HongShi10 — APPROVED** [review]
LGTM, request changes done

**Pull request merged**